app/models/service_template_ansible_playbook.rb

```diff
@@ -69,14 +68,25 @@ def create_new_dialog(dialog_name, job_template, hosts)
   end
   private :create_new_dialog
 
-  def self.create_job_templates(service_name, description, config_info, auth_user)
+  def self.create_job_templates(service_name, description, config_info, auth_user, service_template = nil)
     [:provision, :retirement, :reconfigure].each_with_object({}) do |action, hash|
-      next unless config_info[action] && config_info[action].key?(:playbook_id)
-      hash[action] = { :configuration_template => create_job_template("miq_#{service_name}_#{action}", description, config_info[action], auth_user) }
+      next unless new_job_template_required(config_info[action], action, service_template)
+      hash[action] = { :configuration_template => create_job_template(build_name(service_name, action), description, config_info[action], auth_user) }
     end
   end
   private_class_method :create_job_templates
 
+  def self.new_job_template_required(info, action, service_template)
```

Comment:
`self.new_job_template_required?`

Can you move this method down next to instance method `new_dialog_required?`